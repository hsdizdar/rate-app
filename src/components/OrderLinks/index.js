import React, { useState } from "react";
import { orderByMostVote, orderByLessVote } from "../../utils";

const OrderLinks = ({ parentAddCallback }) => {
  const [value, setValue] = useState("");

  const onChange = (e) => {
    let sortedLinks = [];
    const links = JSON.parse(localStorage.getItem("linkInfo"));
    const orderType = e.target.value;

    localStorage.setItem("orderType", orderType);
    setValue(orderType);

    if (orderType === "asc") {
      sortedLinks = orderByMostVote(links);
    } else if (orderType === "desc") {
      sortedLinks = orderByLessVote(links);
    }

    localStorage.setItem("linkInfo", JSON.stringify(sortedLinks));

    parentAddCallback(true);
  };

  return (
    <div className="form-group">
      <select value={value} onChange={onChange} className="form-control">
        <option hidden>Order By</option>
        <option value="asc">Most Voted (Z - A)</option>
        <option value="desc">Less Voted (A - Z)</option>
      </select>
    </div>
  );
};

export default OrderLinks;
