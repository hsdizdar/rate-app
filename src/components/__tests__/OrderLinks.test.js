import { shallow } from "enzyme";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import OrderLinks from "../OrderLinks";

configure({ adapter: new Adapter() });

let component;

beforeEach(() => {
  component = shallow(<OrderLinks />);
});

describe("OrderLinks rendering", () => {
  it("renders correctly", () => expect(component).toMatchSnapshot());
});
