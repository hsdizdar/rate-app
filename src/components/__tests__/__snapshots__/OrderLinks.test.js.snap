// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderLinks rendering renders correctly 1`] = `ShallowWrapper {}`;
